src/Core/Manifest/ModuleManifest.php

```diff
@@ -225,4 +238,31 @@ public function getModules()
     {
         return $this->modules;
     }
+
+    /**
+     * Get modules sorted by priority
+     *
+     * @return Module[]
+     */
+    public function getSortedModules()
+    {
+        if (!$this->sortedModules) {
+            $order = static::config()->uninherited('module_priority');
+            $project = static::config()->get('project');
+
+            $sorter = Injector::inst()->createWithArgs(
+                PrioritySorter::class.'.modulesorter',
+                [
+                    $this->modules,
+                    $order
+                ]
+            )
+            ->setVariable(self::PROJECT_KEY, $project)
+            ->setDefaultTop(self::PROJECT_KEY);
```

Comment:
If we remove `defaultTop`, then there is a scenario in which you can define `project`, and its effect would be meaningless. Currently what makes `project` special is that even if you don't specify its priority, it will ensure that it comes out on top. Maybe that's a good thing. I like the idea of being explicit about the ordering.

With regard to `$other_modules`, I very much wanted to do that, but I feared it would be met with resistance as an API change. Do you see it as one?